src/NuGet.Core/NuGet.Configuration/Settings/Items/PackageSourceNamespacesItem.cs

```diff
@@ -0,0 +1,233 @@
+
+        /// <summary>
+        /// List of namespaces items part of this package source namespace element.
+        /// </summary>
+        public IList<NamespaceItem> Namespaces { get; }
+
+        public override string ElementName => ConfigurationConstants.PackageSourceAttribute;
+
+        /// <summary>
+        /// Each PackageSourceNamespaces item needs a key.
+        /// The key should correspond a package source key.
+        /// </summary>
+        public virtual string Key => Attributes[ConfigurationConstants.KeyAttribute];
+
+        protected override IReadOnlyCollection<string> RequiredAttributes { get; } = IReadOnlyCollectionUtility.Create(ConfigurationConstants.KeyAttribute);
+
+        protected void SetKey(string value)
+        {
+            if (string.IsNullOrEmpty(value))
+            {
+                throw new ArgumentException(string.Format(CultureInfo.CurrentCulture, Resources.PropertyCannotBeNullOrEmpty, nameof(Key)));
+            }
+
+            UpdateAttribute(ConfigurationConstants.KeyAttribute, value);
+        }
+
+        internal readonly IEnumerable<SettingBase> _parsedDescendants;
+
+        /// <summary>
+        /// Creates a package source namespace item with the given name, which equals the key and non-empty list of naemspace items.
+        /// </summary>
+        /// <param name="name">A non-empty name of the item which corresponds a package source name.</param>
+        /// <param name="namespaceItems">A non empty list of namespace items.</param>
+        /// <exception cref="ArgumentException">If <paramref name="name"/> is null or empty, or <paramref name="namespaceItems"/> is null or empty.</exception>
+        public PackageSourceNamespacesItem(string name, IEnumerable<NamespaceItem> namespaceItems)
+            : base()
+        {
+            if (string.IsNullOrEmpty(name))
+            {
+                throw new ArgumentException(Resources.Argument_Cannot_Be_Null_Or_Empty, nameof(name));
+            }
+
+            if (namespaceItems == null || !namespaceItems.Any())
+            {
+
+                throw new ArgumentException(string.Format(CultureInfo.CurrentCulture, Resources.Error_ItemNeedsAtLeastOneNamespace, name));
+            }
+
+            AddAttribute(ConfigurationConstants.KeyAttribute, name);
+
+            Namespaces = new List<NamespaceItem>();
+
+            foreach (var @namespace in namespaceItems)
+            {
+                Namespaces.Add(@namespace);
+            }
+        }
+
+        internal PackageSourceNamespacesItem(XElement element, SettingsFile origin)
+            : base(element, origin)
+        {
+            _parsedDescendants = element.Nodes().Where(n => n is XElement || n is XText text && !string.IsNullOrWhiteSpace(text.Value))
+                .Select(e => SettingFactory.Parse(e, origin));
+
+            var parsedNamespaceItems = _parsedDescendants.OfType<NamespaceItem>().ToList();
+
+            if (parsedNamespaceItems.Count == 0)
```

Comment:
Would it be perf better?

```suggestion
            if (!parsedNamespaceItems.Any())
```

Reply:
Not sure I follow, why would perf be better with Any?

Reply:
I thought the CA1827 analyzer rule could apply for this case. That's why I suggested using Any

https://docs.microsoft.com/en-us/dotnet/fundamentals/code-analysis/quality-rules/ca1827

Reply:
Ah, that only applies to the `Count()` method, not the count property. 

In fact, `Any()` uses an optimization that specifically checks for Count.
https://github.com/dotnet/runtime/blob/main/src/libraries/System.Linq/src/System/Linq/AnyAll.cs#L18-L21